File: Structural/Facade/DIY.java
package Structural.Facade;

public class DIY {
    public static void main(String[] args) {
        var service = new PlaylistService();
        service.send("My Tears Ricochet", "Taylor Swift");
    
    }
    
}

class Song{
    private String song;
    public Song(String song) {
            this.song = song;
    }

    public String getSong() {
        return song;
    }

}

class Artist{
    private String artist;
    public Artist(String artist){
        this.artist = artist;
    }

    public String getArtist() {
        return artist;
    }
}

class Playlist{

    public Connection connect(int Id){
            return new Connection();
        
    }
    public void play(Song song, Artist artist){
        System.out.println("Playing: " + song.getSong() + " by " + artist.getArtist());
    }

    public void recent(Song song, Artist artist){
        System.out.println("Recently Played: " + song.getSong() + " by " + artist.getArtist());
    }



}

class Connection{
    public void stopPlaying(){
        System.out.println("Playlist stopped.");
    }
}


class PlaylistService{
    public void send(String song, String artist){
        var server = new Playlist();
        var connection = server.connect(12);
        server.connect(1);
        server.play(new Song(song),new Artist(artist));
        connection.stopPlaying();
       
    }

}

File: TS/Structural/Facade/DIY.ts
class Song {
  private song: string;

  constructor(song: string) {
    this.song = song;
  }

  getSong(): string {
    return this.song;
  }
}

class Artist {
  private artist: string;

  constructor(artist: string) {
    this.artist = artist;
  }

  getArtist(): string {
    return this.artist;
  }
}

class MusicPlaylist {
  connect(id: number): Connection {
    return new Connection();
  }

  play(song: Song, artist: Artist): void {
    console.log(`Playing: ${song.getSong()} by ${artist.getArtist()}`);
  }

  recent(song: Song, artist: Artist): void {
    console.log(`Recently Played: ${song.getSong()} by ${artist.getArtist()}`);
  }
}

class Connection {
  stopPlaying(): void {
    console.log("Playlist stopped.");
  }
}

class PlaylistService {
  send(song: string, artist: string): void {
    const server = new MusicPlaylist();
    const connection = server.connect(12);
    server.connect(1);
    server.play(new Song(song), new Artist(artist));
    connection.stopPlaying();
  }
}

class DIYFacade {
  public static main(): void {
    const service = new PlaylistService();
    service.send("My Tears Ricochet", "Taylor Swift");
  }
}

DIYFacade.main();
